rustls-mbedpki-provider/src/client_cert_verifier.rs

```diff
@@ -92,11 +133,26 @@ impl ClientCertVerifier for MbedTlsClientCertVerifier {
             .into_iter()
             .collect();
 
-        verify_certificates_active(chain.iter().map(|c| &**c), now)?;
+        verify_certificates_active(chain.iter().map(|c| &**c), now, self.ignore_expired)?;
 
         let mut error_msg = String::default();
-        mbedtls::x509::Certificate::verify(&chain, &self.trusted_cas, None, Some(&mut error_msg))
-            .map_err(|e| mbedtls_err_into_rustls_err_with_error_msg(e, &error_msg))?;
+        match &self.verify_callback {
+            Some(callback) => {
+                let callback = callback.clone();
+                mbedtls::x509::Certificate::verify_with_callback(
+                    &chain,
+                    &self.trusted_cas,
+                    None,
+                    Some(&mut error_msg),
+                    move |cert: &mbedtls::x509::Certificate, depth: i32, flags: &mut mbedtls::x509::VerifyError| {
+                        callback(cert, depth, flags)
+                    },
+                )
+                .map_err(|e| mbedtls_err_into_rustls_err_with_error_msg(e, &error_msg))?;
+            }
+            None => mbedtls::x509::Certificate::verify(&chain, &self.trusted_cas, None, Some(&mut error_msg))
+                .map_err(|e| mbedtls_err_into_rustls_err_with_error_msg(e, &error_msg))?,
```

Comment:
Can this line (and line 151) be moved outside of `match {...}`?

Reply:
done!